text/0000-layout-packed-aligned.md

```diff
@@ -0,0 +1,197 @@
+- Feature Name: `layout_packed_aligned`
+- Start Date: 2024-10-24
+- RFC PR: [rust-lang/rfcs#3718](https://github.com/rust-lang/rfcs/pull/3718)
+- Rust Issue: [rust-lang/rust#100743](https://github.com/rust-lang/rust/issues/100743)
+
+# Summary
+[summary]: #summary
+
+This RFC makes it legal to have `#[repr(C)]` structs that are:
+- Both packed and aligned.
+- Packed, and transitively contains`#[repr(align)]` types.
+
+It also introduces `#[repr(system)]` which is designed for interoperability with operating system APIs.
+It has the same behavior as `#[repr(C)]` except on `*-pc-windows-gnu` targets where it uses the msvc layout
+rules instead.
+
+# Motivation
```

Comment:
Can we explicitly document that a goal of this RFC is to make sure that existing uses of `#[repr(C)]` are required to clarify their intent (ie, to indicate compatibility with a C compiler or to indicate a "linear" layout)? In particular, this implies that `#[repr(C)]` itself should be deprecated, and the two behaviors should both be supported by new reprs. That will ensure that code doesn't silently do either of the following:
- Change its behavior (if we decide to change `#[repr(C)]` to mean "compatible with the C compiler")
- Remain subtly buggy (if we decide to keep `#[repr(C)]`'s behavior, and that code was intended to be compatible with the C compiler)